File: apps/frontend/src/types/product.ts
export type Product = {
  id: string
  vendorId: string
  categoryId: string
  name: string
  description: string
  slug: string
  isActive: boolean
  createdAt: string
  updatedAt: string
  vendorName: string
  categoryName: string
}

export type ProductVariant = {
  id: string
  sku: string
  price: number
  stock: number
  attributes?: Record<string, string>
}

export type ProductAttribute = {
  name: string
  values: string[]
}

export type ProductDetails = {
  id: string
  name: string
  slug: string
  description: string
  categoryId: string
  vendorId?: string
  categoryName: string
  vendorName: string
  isActive: boolean
  variants: ProductVariant[]
  attributes: ProductAttribute[]
  imageUrls: string[]
}

File: apps/backend/EcommerceApi/Models/Product.cs
using System.ComponentModel.DataAnnotations;

namespace EcommerceApi.Models
{
    public class Product
    {
        public Guid Id { get; set; }

        public Guid? VendorId { get; set; }
        public Guid CategoryId { get; set; }

        [Required, MaxLength(200)]
        public string Name { get; set; } = string.Empty;

        public string? Description { get; set; }

        public bool IsActive { get; set; } = true;

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
        // 🆕 SEO-friendly identifier
        [Required, MaxLength(200)]
        public string Slug { get; set; } = string.Empty;
        // Navigation
        public Vendor? Vendor { get; set; }
         public Category? Category { get; set; }
        public ICollection<ProductVariant>? Variants { get; set; }
        public ICollection<ProductImage>? Images { get; set; }
        public ICollection<Review>? Reviews { get; set; }
    }
}
